### PR TITLE
refactor: Sync inline nodes code with the structure of block nodes [DANTE-1167]

### PR DESCRIPTION
**Pull request by msieroslawska**
This refactoring PR is the **first part** of introducing embedded inline cross-space links and is supposed to make the introduction easier. Once we have a similar code structure for block and inline nodes, the effort of adding inline resource links should be lower.

There are no functionality changes in the scope of this PR, only renaming components and moving the code around.

### CODE REVIEW

File: packages/rich-text/src/plugins/shared/LinkedInlineWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  };
+}>;
+
+export function LinkedInlineWrapper({
+  attributes,
+  card,
+  children,
+  element,
+}: LinkedInlineWrapperProps) {
+  const link = element.data.target;
```
Comment (Silhoue) [inline packages/rich-text/src/plugins/shared/LinkedInlineWrapper.tsx:44]: Could we only pass the `target` down if that's all we need? Or will we need more later on?

**Review by Silhoue — APPROVED** [review]
Great that you're aiming to split it into smaller PRs 👍

File: packages/rich-text/src/plugins/shared/LinkedInlineWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    display: 'inline-block',
+    margin: `0 ${tokens.spacing2Xs}`,
+    fontSize: 'inherit',
+    span: {
+      userSelect: 'none',
+    },
+  }),
+};
+
+const isResourceLink = (link: EntryLink | ResourceLink): link is ResourceLink =>
```
Comment (chrishelgert) [inline packages/rich-text/src/plugins/shared/LinkedInlineWrapper.tsx:25]: I think we have this one also in the LinkedBlockWrapper, can we maybe extract it to an util?
Or maybe even this snipped `isResourceLink(link) ? link.sys.urn : link.sys.id` as a function

**Review by chrishelgert — APPROVED** [review]
Nice cleanup 👍

**Pull request merged**